src/main/java/de/ustutt/iaas/cc/core/textProcessors/loadBalancerStrategies/RoundRobin.java

```diff
@@ -0,0 +1,41 @@
+package de.ustutt.iaas.cc.core.textProcessors.loadBalancerStrategies;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.ws.rs.client.WebTarget;
+
+import org.slf4j.Logger;
+
+/**
+ * Load balancer strategy which chooses the server alternately and sends the
+ * request to it.
+ *
+ */
+public class RoundRobin extends AbstractLoadBalancerStrategy {
+	private static final Logger logger = org.slf4j.LoggerFactory.getLogger(RoundRobin.class);
+
+	private AtomicInteger index = new AtomicInteger();
+	private static List<WebTarget> targets;
+
+	public RoundRobin(List<WebTarget> targets) {
+		RoundRobin.targets = targets;
+		logger.debug("Number of Web Targets alias TextProcessors: {}", targets.size());
+	}
+
+	@Override
+	protected WebTarget getNextTarget() {
+		if (index.get() == targets.size()) {
+			index.set(0);
```

Comment:
This will not work in any case. In cases of high concurrency there may be situations where you skip the if-statement but when you call the `getAndIncrement()` method, you will get a value that is bigger than the number of targets (because parallel executions incremented the index counter in the meantime).

Reply:
True :). In the SoapUI project I've increased the number of threads up to 20 and the index was higher than the number of targets.